ansys/dpf/core/server_context.py

```diff
@@ -0,0 +1,101 @@
+"""
+ServerContext
+=============
+
+Gives the ability to choose the context with which the server should be started.
+The context allows to choose which capabilities are available.
+"""
+from enum import Enum
+
+
+class EContextType(Enum):
+    pre_defined_environment = 0
+    """DataProcessingCore.xml that is next to DataProcessingCore.dll/libDataProcessingCore.so will
+    be taken"""
+    premium = 1
+    """Gets the Specific premium DataProcessingCore.xml."""
+    user_defined = 2
+    """Load a user defined xml using its path."""
+    custom_defined = 3
+    """Loads the xml named "DpfCustomDefined.xml" that the user can modify."""
+    entry = 4
+    """Loads the minimum number of plugins for a basic usage."""
+
+
+class ServerContext:
+    """The context allows to choose which capabilities are available server side.
+
+    The default parameters can be overwritten using the DPF_SERVER_TYPE environment
+    variable. DPF_SERVER_TYPE=INPROCESS, DPF_SERVER_TYPE=GRPC,
+    DPF_SERVER_TYPE=LEGACYGRPC can be used.
+
+    Parameters
+    ----------
+    context_type : EContextType
+        Type of context.
+    xml_path : str, optional
+        Path to the xml to load.
+    """
+    def __init__(self, context_type=EContextType.user_defined, xml_path=""):
```

Comment:
@cbellot000 Why is the user_defined context the default one? I think this makes the xml_path non-optional in a way.

Reply:
@PProfizi it's the default of this constructor (because you would only need to use the constructor directly when you want to speicify an xml), it doesn't make user defined the default context (which is SERVER_CONTEXT = AvailableServerContexts.entry)